src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java

```diff
@@ -146,12 +151,16 @@ private void updateReadingGoal(int targetToRead, ReadingGoal.GoalType goalType)
 
         if (goalType.equals(ReadingGoal.GoalType.BOOKS)) {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + + targetToRead + getPluralized(" book", targetToRead));
-            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
             booksToRead.setText(calculateBooksToRead(targetToRead, howManyReadThisYear));
+            togglePagesGoalInfo(false, hasReachedGoal);
             toggleBooksGoalInfo(true, hasReachedGoal);    // show book goal-specific information
         } else {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + targetToRead + getPluralized(" page", targetToRead));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
+            booksToRead.setText(calculateBooksToRead(targetToRead, howManyReadThisYear));
```

Comment:
You're using `booksToRead` here while the goal type is pages. If you still want to use that variable, I'd recommend renaming it and also considering whether you want to rename the similarly named `pagesToRead`